### DISCUSSION TITLE
Relationship triggers do not cascade...

### DISCUSSION THREAD

**Discussion by AlexVanK** [directus/directus#10169] [Archive]
Hello,

I have a few workflow complications here:

1. Relational Triggers don't exist for Files interface
1. Relational triggers are not inherited more than 1 level deep
1. How to force relational triggers in hooks?

---

1. 
Not sure if this is the intended option but there's no relationship trigger for the Files relationship interface.

Normally a regular relationship field has these options:
![image](https://user-images.githubusercontent.com/56647386/144103573-abe17105-8fc7-45a7-9b67-8a5bb07c0991.png)

How do I go about having the same for Files fields?

---

2. 

USE Case:

- I have a "Supporting Documentation" relationship field in my main collection.
- Upon deleting one Supporting Doc entry from my main application, the related item is removed from the related collection.
BUT
- The related collection also contains a relational field "Files" which hold the actual file. I would like this also to inherit the deletion actions as specified in the picture above.

In other words, if I have an arbitrary deeply nested (through relationship fields) collection structure such that:

A > B > C > D

if an element B is removed from A, whatever action is decided upon the removal, that action should propagate in C and D.
Otherwise in the use case expressed above, we could run into orphan files that have no more purpose...

---

3.

How do I handle this in the hooks?
Can I just nullify a relationship to fire the relationship trigger,
or do I have to manually follow the logic down to the very last nested item, and manually enforce the desired status?

**Comment by rijkvanzanten** [comment]
> Relational Triggers don't exist for Files interface

That doesn't sound right to me. The files interface is just a regular m2o which pre-fills the relationship to `directus_files`, and therefore should have the relational constraints as well. Mind opening a new issue for that?

> Relational triggers are not inherited more than 1 level deep

I don't think this is true 🤔 I think you might run into a misunderstanding of how the relational constraints operate. The constraints are the foreign key constraint triggers from the database directly, and operate in the same way. This means that they're always operated from the "perspective" of the many to one side of the relationship.

> How to force relational triggers in hooks?

There's nothing special to do in hooks. The relational triggers exist on the database, and will fire whenever the database is changed in a way that would trigger the constraint.
